libraries/ms-common/src/x86/i2c.c

```diff
@@ -1,23 +1,63 @@
 #include "i2c.h"
+#include "log.h"
+#include "stdio.h"
 
 StatusCode i2c_init(I2CPort i2c, const I2CSettings *settings) {
-  return STATUS_CODE_UNIMPLEMENTED;
+  if (i2c >= NUM_I2C_PORTS) {
+    return status_msg(STATUS_CODE_INVALID_ARGS, "Invalid I2C port.");
+  } else if (settings->speed >= NUM_I2C_SPEEDS) {
+    return status_msg(STATUS_CODE_INVALID_ARGS, "Invalid I2C speed.");
+  }
+  LOG_DEBUG("Note this is an x86 version of I2C\n");
+  return STATUS_CODE_OK;
 }
 
 StatusCode i2c_read(I2CPort i2c, I2CAddress addr, uint8_t *rx_data, size_t rx_len) {
-  return STATUS_CODE_UNIMPLEMENTED;
+  if (i2c >= NUM_I2C_PORTS) {
+    return status_msg(STATUS_CODE_INVALID_ARGS, "Invalid I2C port.");
+  }
+  LOG_DEBUG("Reading %ld bytes over I2C\n", rx_len);
+  for (size_t i = 0; i < rx_len; i++) {
+    // Insert dummy data
+    rx_data[i] = i % 2;
+  }
+  return STATUS_CODE_OK;
 }
 
 StatusCode i2c_write(I2CPort i2c, I2CAddress addr, uint8_t *tx_data, size_t tx_len) {
-  return STATUS_CODE_UNIMPLEMENTED;
+  if (i2c >= NUM_I2C_PORTS) {
+    return status_msg(STATUS_CODE_INVALID_ARGS, "Invalid I2C port.");
+  }
+  LOG_DEBUG("Sending %ld bytes over I2C\n", tx_len);
+  for (size_t i = 0; i < tx_len; i++) {
+    printf("0x%x ", tx_data[i]);
+  }
+  printf("\n");
```

Comment:
Shoot, I just realized... this is going to break filtering the verbosity of the log. Could you surround the for loop and printf with `if (LOG_LEVEL_DEBUG >= LOG_LEVEL_VERBOSITY) {..}` please? That way the printfs will behave nicely like `LOG_DEBUG`.

(Also, could you change the `\n` in the `LOG_DEBUG("Sending %ld bytes over I2C\n")` to "`: `" so that the whole message is on one line?)